define/types.go

```diff
@@ -101,11 +101,43 @@ type Secret struct {
 
 // BuildOutputOptions contains the the outcome of parsing the value of a build --output flag
 type BuildOutputOption struct {
+	Type     BuildOutputType
+	Attrs    map[string]string
 	Path     string // Only valid if !IsStdout
 	IsDir    bool
 	IsStdout bool
 }
 
+type BuildOutputType int
+
+const (
```

Comment:
> suggests that the external string parser, and the internal representation, may very well be different.

It's a shortcut yes, but we can very well still go ahead and use `--type=image` and it should work as expected or handle the flag appropriately. Not sure if we can differentiate this as interval vs external? Or I might be missing something.

Reply:
> Is this intended to match docker buildx --output?
> 
> We might only need that compatibility in the option parser, not in the enum (yet?).

Yes, that's the plan but adding the not-yet-implemented constants shouldn't make too much of a difference, right? But if it's a matter of hygiene, then it makes sense.

Reply:
In general the way I think about this is that the buildx syntax is a CLI-level concern that’s not really important to the lower levels. (The push not to represent `Attrs` as raw strings is both an example of that philosophy, and a demonstration why using native data can be more useful.)

Notably if some feature isn’t implemented, the CLI layer should cleanly fail instead of doing the wrong thing — and that’s easier if the unimplemented features can’t be represented at all, instead of having a remote invisible dependency between the low-level implementation and the high-level CLI parser recognizing exactly the same subset of features.